app/static/js/detail_tender_view.js

```diff
@@ -1,46 +1,38 @@
 
 function pushNewTagToTable(tagInputValue) {
-    const tagsList = document.querySelector('#tags_list_displayed')
-    const tagsListValues = tagsList.innerText.split(',')
+    const tagsList = $("#tags_list_displayed")
+    const tagsListValues = tagsList.text().split(',')
 
-    tagsListValues.push(tagInputValue)
+    const trimmedArray = tagsListValues.map(str => str.trim());
+    if (!trimmedArray.includes(tagInputValue.trim())) {
+        tagsListValues.push(tagInputValue)
 
-    const filteredArr = tagsListValues.filter(function (element) {
-        return element !== '';
-    });
+        const filteredArr = tagsListValues.filter(function (element) {
+            return element !== '';
+        });
 
-    tagsList.innerHTML = filteredArr.join(', ')
-}
-
-
-
-
-function pushTagDb(id, tagInputValue) {
-    console.log(id)
-    $.ajax({
-        type: 'POST',
-        url: `/tenders/tag/${id}/`,
-        data: {
-            tag_name: tagInputValue,
-            csrfmiddlewaretoken: getCookie('csrftoken')
-        },
-        success: function (response) {
-            console.log(response)
-            document.querySelector('#tagValue').value = ''
+        tagsList.html(filteredArr.join(', '))
+    }
 
-        }
-    });
 }
 
+$('#add_tag_button').click(function () {
```

Comment:
This should be wrapped inside a `$(document).ready()` block: https://learn.jquery.com/using-jquery-core/document-ready/

Reply:
done